timelock-agent/src/main/java/com/palantir/timelock/paxos/LockCreator.java

```diff
@@ -17,33 +17,31 @@
 package com.palantir.timelock.paxos;
 
 import java.util.concurrent.Semaphore;
+import java.util.function.Supplier;
 
 import com.palantir.atlasdb.timelock.lock.BlockingTimeLimitedLockService;
 import com.palantir.lock.CloseableLockService;
 import com.palantir.lock.LockServerOptions;
 import com.palantir.lock.impl.LockServiceImpl;
 import com.palantir.lock.impl.ThreadPooledLockService;
-import com.palantir.timelock.Observables;
 import com.palantir.timelock.config.TimeLockDeprecatedConfiguration;
 import com.palantir.timelock.config.TimeLockRuntimeConfiguration;
 
-import io.reactivex.Observable;
-
 public class LockCreator {
-    private final Observable<TimeLockRuntimeConfiguration> runtime;
+    private final Supplier<TimeLockRuntimeConfiguration> runtime;
     private final TimeLockDeprecatedConfiguration deprecated;
 
     private static Semaphore sharedThreadPool = new Semaphore(-1);
 
-    public LockCreator(Observable<TimeLockRuntimeConfiguration> runtime, TimeLockDeprecatedConfiguration deprecated) {
+    public LockCreator(Supplier<TimeLockRuntimeConfiguration> runtime, TimeLockDeprecatedConfiguration deprecated) {
         this.runtime = runtime;
         this.deprecated = deprecated;
     }
 
     public CloseableLockService createThreadPoolingLockService() {
         // TODO (jkong): Live reload slow lock timeout, plus clients
```

Comment:
Surely there's a github issue that we can attach to this TODO.

Reply:
Interestingly, I didn't find one (searched "reload"), but here we are: #2342 

Also tagged this TODO with the ticket.